ql/src/java/org/apache/hadoop/hive/ql/QueryCompilationBreakdownSummary.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.ql;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.hadoop.hive.ql.exec.tez.monitoring.PrintSummary;
+import org.apache.hadoop.hive.ql.log.PerfLogger;
+import org.apache.hadoop.hive.ql.session.SessionState;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.hadoop.hive.ql.exec.tez.monitoring.Constants.SEPARATOR;
+
+public class QueryCompilationBreakdownSummary implements PrintSummary {
+  private static final String OPERATION_SUMMARY = "%-84s %9s";
+  private static final String QUERY_COMPILATION_SUMMARY = "Query Compilation Summary";
+  private static final String LINE_SEPARATOR = "line.separator";
+  private static final String TOTAL_COMPILATION_TIME = "Total Compilation Time";
+  private List<Pair<String, String>> compileSteps = new ArrayList<>();
+
+  private String format(String value, long number) {
+    return String.format(OPERATION_SUMMARY, value, number + "ms");
+  }
+
+  private void populateCompileStepNumbers() {
```

Comment:
The content is immutable so we don't need to populate the `compileSteps` on every call. The whole thing can be become static final and initialized only once.